tests/testtransport.nim

```diff
@@ -30,14 +30,18 @@ suite "TCP transport":
   test "test listener: handle read":
     proc testListener(): Future[bool] {.async.} =
       let ma: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+      var finished = newAsyncEvent()
       proc connHandler(conn: Connection): Future[void] {.async, gcsafe.} =
         let msg = await conn.read(6)
         check cast[string](msg) == "Hello!"
+        finished.fire()
 
       let transport: TcpTransport = newTransport(TcpTransport)
-      asyncCheck await transport.listen(ma, connHandler)
+      asyncCheck transport.listen(ma, connHandler)
       let streamTransport: StreamTransport = await connect(transport.ma)
       let sent = await streamTransport.write("Hello!", 6)
+      await finished.wait()
+      await transport.close()
```

Comment:
`streamTransport` is leaking here.

Reply:
could you elaborate?

Reply:
yes, check my PR,  it has to be closed indeed, or tracker check would fail